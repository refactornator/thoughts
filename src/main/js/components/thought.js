import React from 'react';
import moment from 'moment';
import ActionDelete from 'material-ui/svg-icons/action/delete';
import IconButton from 'material-ui/IconButton';
import {Card, CardHeader, CardText} from 'material-ui/Card';

export default class Thought extends React.Component {
    render() {
        const style = Object.assign({}, {marginTop: 10});
        const sinceHappened = moment(this.props.thought.creationTime).fromNow();
        return (
            <Card style={style}>
                <CardHeader subtitle={sinceHappened} style={{height: 20}}>
                    <IconButton className="delete-button" iconStyle={{color: 'rgba(0, 0, 0, 0.541176)', width: 20, height: 20}}>
                        <ActionDelete onClick={this.props.deleteThoughtHandler} />
                    </IconButton>
                </CardHeader>
                <CardText>
                    {this.props.thought.text}
                </CardText>
            </Card>
        )
    }
}