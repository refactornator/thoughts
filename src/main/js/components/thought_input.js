import React from 'react';
import { connect } from 'react-redux';
import Toggle from 'material-ui/Toggle';
import { Card, CardText } from 'material-ui/Card';
import RaisedButton from 'material-ui/RaisedButton';
import LocalStorage from '../utilities/local_storage';

import { createThought, changeThought, changeCategory } from '../actions';

const styles = {
  label: {
    float: 'left',
    marginLeft: 30,
    marginRight: 10,
    marginTop: 8,
    color: 'rgba(0, 0, 0, 0.870588)'
  },
  toggle: {
    marginTop: 7,
    width: null,
    display: 'inline-block',
    marginBottom: 16
  }
};

class ThoughtInput extends React.Component {
  handleNewThought() {
    this.props.createThought({
      category: this.props.category,
      text: this.props.thought
    });
  }

  handleToggleChange(event, toggled) {
    const newCategory = toggled ? 'life' : 'work';
    this.props.changeCategory(newCategory);
  }

  render() {
    const { category, thought, changeThought } = this.props;
    return (
      <Card className="thought-input" style={{ zIndex: 100 }}>
        <CardText style={{ padding: '9px 16px 0 16px' }}>
          <textarea
            value={thought}
            onChange={event => changeThought(event.target.value)}
          />
          <label style={styles.label}>Work</label>
          <Toggle
            label="Life"
            toggled={category === 'work' ? false : true}
            labelPosition="right"
            onToggle={this.handleToggleChange.bind(this)}
            style={styles.toggle}
          />
          <div style={{ textAlign: 'right', float: 'right' }}>
            <RaisedButton
              className="remember-button"
              primary={true}
              label="Remember"
              onClick={this.handleNewThought.bind(this)}
            />
          </div>
        </CardText>
      </Card>
    );
  }
}

export default connect(({ category, thought }) => ({ category, thought }), {
  createThought,
  changeThought,
  changeCategory
})(ThoughtInput);
