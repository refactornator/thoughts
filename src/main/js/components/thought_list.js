import React from 'react';

import Thought from './thought';

export default class ThoughtList extends React.Component {
    render() {
        return (
            <ul className="thoughts">
                {this.props.thoughts.map(({key, style, data}) => {
                    let deleteThoughtHandler = this.props.deleteThoughtHandler.bind(this, data._links.self.href);
                    return <Thought
                        key={key}
                        style={style}
                        thought={data}
                        deleteThoughtHandler={deleteThoughtHandler}
                    />;
                })}
            </ul>
        )
    }
}
