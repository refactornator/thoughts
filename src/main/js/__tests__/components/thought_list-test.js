import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
jest.mock('../../components/thought');
import { ThoughtList } from '../../components/thought_list';

describe('ThoughtList', () => {
  it('renders a list of thoughts correctly', () => {
    const thoughts = [
      {
        id: 2,
        category: 'life',
        text: 'This is a thought.',
        creationTime: new Date().toISOString(),
        _links: {
          self: {
            href: 'http://example.com/page/2'
          }
        }
      }
    ];
    const requestThoughtsMock = jest.fn();
    const renderedThoughtList = renderer
      .create(
        <MuiThemeProvider>
          <ThoughtList
            category="life"
            thoughts={thoughts}
            requestThoughts={requestThoughtsMock}
          />
        </MuiThemeProvider>
      )
      .toJSON();
    expect(requestThoughtsMock).toBeCalled();
    expect(renderedThoughtList).toMatchSnapshot();
  });
});
