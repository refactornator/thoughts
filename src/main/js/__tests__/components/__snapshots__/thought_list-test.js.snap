// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThoughtList renders a list of thoughts correctly 1`] = `
<ul
  className="thoughts"
>
  <span />
</ul>
`;
